rendering-event-loop.md

```diff
@@ -0,0 +1,94 @@
+### Event loops
+
+HTML and script in web pages use an event loop-driven model of computation. The event loops for HTML are defined [here](https://html.spec.whatwg.org/multipage/webappapis.html#event-loop). Tasks on an event loop live in one or more [task queues](https://html.spec.whatwg.org/multipage/webappapis.html#task-queue), and each loop only has one task at a time. Event loops also have [microtask queues](https://html.spec.whatwg.org/multipage/webappapis.html#microtask), which are generally processed all-at-once at the end of each time script may run, or at the end of tasks. These are called [microtask checkpoints](https://html.spec.whatwg.org/#perform-a-microtask-checkpoint).
+
+While there can only be one task at a time on an event loop, tasks can execute
+[in parallel](https://html.spec.whatwg.org/#parallelism) in some cases, and also generate promises to mark their completion. These promises, which can be held onto by script, run completion microtasks (see also [this spec text](https://html.spec.whatwg.org/#integration-with-the-javascript-job-queue)) when the underlying (async) task is complete.
+
+### The browser context (aka rendering) event loop
+
+There are several event loops, but the most important is the
+[browsing context](https://html.spec.whatwg.org/multipage/browsers.html#browsing-context) event loop. This event loop is special because it is where tasks related to
+* document-loaded scripts (as opposed to worker scripts),
+* DOM,
+* input event processing
+* rendering updates
+
+all occur.
+
+As it relates to rendering, the spec currently defines image decoding, video,
+and HTMLCanvasElement.{toBlob,convertToBlob} to be in parallel.
+
+However, any task which affects script-observable data (i.e. specific to a
+particular script realm, global or evironment settings object) [must not be in parallel](https://html.spec.whatwg.org/#event-loop-for-spec-authors) to the task queue on which such a script runs. This means that:
```

Comment:
"such a script" is referring to the task? ie any task which affects script-observable data must not run in parallel with other tasks in the same queue?

Reply:
> "such a script" is referring to the task? ie any task which affects script-observable data must not run in parallel with other tasks in the same queue?

I adjusted the text to improve clarity:

Reply:
However, any task which affects data observable by a script (i.e. specific to a
particular script realm, global or environment settings object) must not run in parallel with other tasks in the queue which may run the script.